### ISSUE TITLE
SteamOS Latest: Higher render resolution than native + FSR disables when mouse cursor is drawn

### ISSUE THREAD

**Issue by rudemario** [ValveSoftware/gamescope#907]
**Platform: Steam Deck
Version: SteamOS 3.4.8**

Confirmed occuring on 2 separate Steam Decks, one on latest (3.4.8) and one on 3.4.4.

**Issue:** 
When running a game where mouse input is needed and therefore used, and also auto-hides the mouse cursor after a few seconds, and running said game at a higher internal resolution and downscaling using Gamescope through the SteamUI (2560x1600>1280x800), when the FSR Scaling Filter setting is checked, whenever the mouse is drawn on screen the image noticeably sharpens, and whenever the mouse disappears a few seconds later, the image returns to its softer FSR state. In my testing there wasn't a single game this didn't apply to, as long as the game had a hideable cursor for easily noticable A/B testing (RimWorld, CoD: Black Ops II), the only pre-requisite was a higher internal rendered resolution than the output resolution and FSR being selected as the scaling filter.

The image appears to be as sharp, but different in appearance to integer scaling, so my guess is that the mouse cursor being drawn interrupts some portion of the pipeline for the post processing algorithm for FSR but not the entire process, causing an incompletely processed image whenever the mouse is drawn, instead of an outright disabled state.

This issue does not occur with the other scaling options, such as Integer and Nearest Neighbour, likely for obvious reasons.

For now, I am forced to integer scale when supersampling in games where I use the mouse because the softening and edge anti-aliasing provided by FSR 1.0 which improves the supersampled image quality once downscaled and smoothes out edges is constantly flickered on and off and it produces too distracting a play experience.

**Reproduction steps:**

**1.** Identify a game that has a mouse cursor that hides itself either after a few seconds or on controller input (RimWorld and Black Ops II are examples I gave)
**2.** Navigate to said game's Game Properties in the SteamUI and set the resolution to a higher resolution, like 2560x1600 and set the "Internal and External resolution" option to allow up/downscaling on the Steam Deck screen (this happens on ALL higher than internal resolutions and regardless of aspect ratio, like 1920x1080, not just with 2560x1600 and other integer multiples of other aspect ratios)
**3.** Open the Quick Access Menu and navigate to Battery>Scaling Filter and enable FSR (any sharpness setting will do)
**4.** Launch the game and get to a menu where you can make use of step 1.
**5.** Pay close attention to the outlines of boxes or the fringes of text. You should very clearly be able to spot a sharpening of the image when you move the mouse with the trackpads and trigger it being drawn, and then a restoration of the softness of the complete FSR image when you have the mouse cursor hide itself.

**Hypothesis:**
Something about the mouse cursor being drawn is preventing FSR from applying all of its post process adjustments as evidenced by the image still differing when the cursor is being drawn from the other scaling options, instead of being outright disabled by the mouse. Some interruption/selfishness in the graphics pipeline by Linux's mouse drawing?

**Proposal:**
**There is valid reason to want an algorithm like FSR to apply when supersampling.** It softens aliasing and improves text shape via that softening, while the supersampling provides a very strong boost to image quality. Many games that aren't GPU intensive can be supersampled 4x with only a less than 5 FPS drop, meaning many games on the Steam Deck are capable of having this feature applied if the user desires in a reasonable and practical sense. To use RimWorld, a unity game with 2D assets + zooming in, running 4x supersampled is a massive improvement to image clarity. Losing this ability by disabling it would prevent this valid, image quality increasing use case from being used anymore. If possible, can this be fixed instead of FSR being outright disabled when supersampling? **FSR works perfectly fine when supersampling, only failing partially when the mouse cursor is drawn.** **This does not happen when upscaling, which means it's likely possible to fix this behavior when downscaling too,** although I won't presume to know the reality of how the two are handled and differ from each other in the code.

Let me know if there's anything else you need or anything else I can help you with, I'll do my best. I use supersampling whenever I can because it makes a massive difference compared to 720p native so I'd really like to see this fixed one day, and wanna help where I can.

Sidenote: I also tried to see if this occured when running gamescope on Ubuntu as well as on the Steam Deck but the only laptop I have Linux on is Haswell based so I wasted my time, if someone is able to reproduce this on a different distribution that would be great.

Thanks so much!

**Comment by rudemario** [comment]
Correction! This issue also occurs when using the Linear scaling Filter option.

That means the only two unaffected scaling filter options are Nearest and Integer.

I tried out Linear because it also softens the image a bit and lo and behold, same issue.

This means **it's not FSR specifically, but something about Gamescopes execution of both Linear and FSR.**

**Comment by trevyn** [comment]
Also noticed this. Have you tried this with a game that allows forcing a software cursor vs hardware cursor?

**Comment by chomskysfavefive** [comment]
Supersampling is clearly becoming a popular method to experience older games at high-res quality. The merges have been completed and stable forever now. We even have options, bicubic or FSR. What's the issue and how can I help get this over the hump?

**Comment by Isaac-Clarke** [comment]
This issue should be correlated:
https://github.com/ValveSoftware/gamescope/issues/692

tl;dr
Using gamescope for SSAA doesn't work because there's not "FXAAization of the image to blend the pixels together, therefore your GPU works more and the image stays the same"...
